crates/circuit/src/dag_circuit.rs

```diff
@@ -4506,6 +4549,59 @@ impl DAGCircuit {
         Ok(dag_node)
     }
 
+    /// Returns an iterator over all the indices that refer to an `Operation` node in the `DAGCircuit.`
+    pub fn op_nodes<'a>(
+        &'a self,
+        include_directives: bool,
+    ) -> Box<dyn Iterator<Item = NodeIndex> + 'a> {
```

Comment:
I am not keen on using Box unless it is strictly necessary, but we can easily change this later :)